### PR TITLE
feat(dep): add private registry support

### PR DESCRIPTION
**Pull request by opichals**
This removes the use of any npmjs.io API and solely relies on the
npm cli. Therefore it now works for private repo stuff.

```
$ bb dep check @apiaryio/ui-kit
Analyzing the package @apiaryio/ui-kit@latest ...
Getting dependencies...
...
```

### CODE REVIEW

File: blackbelt/dependencies.py [hunk 1]
```diff
@@ -66,8 +66,8 @@ def check(dep, list_path, licenses_path, dev=False, debug=False):
     """
     ensure_executables(['npm', 'license-checker'])
 
-    click.echo('Analyzing the package...')
     dep_name, dep_version = dep
+    click.echo('Analyzing the package '+dep_name+'@'+dep_version+' ...')
```
Comment (honzajavorek) [inline blackbelt/dependencies.py]: Nitpick: whitespace around operators...

**Update:** Well, not. This should actually be:

```python
'Analyzing the package {}@{} ...'.format(dep_name, dep_version)
```

If you are interested in learning the quirks of the formatting, https://pyformat.info/ is a great resource. In Python 3.6+, you could also

```python
f'Analyzing the package {dep_name}@{dep_version} ...'
```

...which is cool, but unfortunately we did not drop Python 2 support on Black Belt yet 🙄

File: blackbelt/commands/dep.py [hunk 2]
```diff
@@ -18,18 +18,6 @@ def validate_dep(ctx, param, dep):
...
-                response.raise_for_status()
-                dep_version = response.json()['collected']['metadata']['version']
-            except requests.HTTPError as e:
-                if e.response.status_code == 404:
-                    raise click.BadParameter('The npm package does not exist')
-                else:
-                    if ctx.params.get('debug'):
-                        raise
-                    raise click.ClickException('Unable to figure out the package version. See --debug for more information.')
         return (dep_name, dep_version)
```
Comment (honzajavorek) [inline blackbelt/commands/dep.py]: You can simplify it to this extent:

```python
     try:
         return parse_dep(dep)
    except Exception:
        raise click.BadParameter('The dependency format should be e.g. react@16.2')
```

**Comment by opichals** [comment]
@honzajavorek 

The version of the `latest` tag is eventually displayed (although after the analysis is done).

```
$ bb dep check bower
Analyzing the package bower@latest ...
Getting dependencies...
Creating the list of 4th party deps... /Users/standa/Dev/black-belt/list.txt
Creating the Public License field contents... /Users/standa/Dev/black-belt/licenses.txt

bower@1.8.4
License: MIT
Copyright Notice: Copyright (c) 2013-present Twitter and other
contributors
Dependencies: 0
Eligible for Pre-Approval: True

Description: The browser package manager
Package: https://npmjs.com/package/bower
Repo: https://github.com/bower/bower
```

The module 404 errors are handled the same way they are for non-existing module version. Output examples follows:


*Missing module*

```
$ bb dep check bower-nonexistent-npm
Analyzing the package bower-nonexistent-npm@latest ...
Getting dependencies...
Usage: bb dep check [OPTIONS] DEP

Error: Invalid value: The npm package could not be installed
```

*Missing module version*

```
$ bb dep check bower@333.33.33
Analyzing the package bower@333.33.33 ...
Getting dependencies...
Usage: bb dep check [OPTIONS] DEP

Error: Invalid value: The npm package could not be installed
```

**Comment by opichals** [comment]
This fixes the #160 finally

```
$ bb dep check @commitlint/cli 
Analyzing the package @commitlint/cli@latest ...
Getting dependencies...
Creating the list of 4th party deps... /Users/standa/Dev/black-belt/list.txt
Creating the Public License field contents... /Users/standa/Dev/black-belt/licenses.txt

@commitlint/cli@7.0.0
License: MIT
Copyright Notice: Copyright (c) Mario Nebl <hello@herebecode.com>
Dependencies: 145
Eligible for Pre-Approval: False
```

**Comment by honzajavorek** [comment]
Do we have a way to debug what caused npm not to be able to install? IMHO the best experience would be:

1. ~`bb dep check <nonexisting>` => `Error: The npm package does not exist`~
2. ~`bb dep check <existing>@<nonexisting>` => `Error: The version does not exist (available: 1.2, 1.1, 1.0...)` (printing last three from `npm view <existing> versions --json`~
...!!! filed https://github.com/apiaryio/black-belt/issues/178 instead. Let's do it later.

**Pull request merged**